extensions/env-injector/src/main.rs

```diff
@@ -0,0 +1,175 @@
+
+    let snap_context = std::env::var("SNAP_CONTEXT")?;
+
+    let request_body = format!(
+        r#"{{"context-id":"{}","args":["get", "env", "envfile", "apps"]}}"#,
+        snap_context
+    );
+
+    let req: Request<Full<Bytes>> = Request::builder()
+        .method(Method::POST)
+        .uri(url)
+        .body(Full::from(request_body))?;
+
+    let mut res = client.request(req).await?;
+
+    let mut body: Vec<u8> = Vec::new();
+
+    while let Some(frame_result) = res.frame().await {
+        let frame = frame_result?;
+
+        if let Some(segment) = frame.data_ref() {
+            body.extend_from_slice(segment);
+        }
+    }
+
+    Ok(serde_json::from_slice(&body)?)
+}
+
+fn process_env(env: &serde_json::Value) -> HashMap<String, String> {
+    let obj = env.as_object()
+        .ok_or("Expected an object (JSON input)").unwrap();
+    let mut map = HashMap::new();
+
+    for (k, v) in obj {
+        if v.is_object() || v.is_array() {
+            eprintln!(
+                "Skipped invalid key containing dots: {}",
+                v
+            );
+            continue;
+        }
+
+        let key = k.to_uppercase().replace("-", "_");
+        let value = v.to_string();
+        map.insert(key, value);
+    }
+    map
+}
+
+
+
+
+fn set_env_vars(app: &str, json: &serde_json::Value) -> Result<(), Box<dyn std::error::Error + Send + Sync>> {
+    let stdout_str = json["result"]["stdout"].as_str().ok_or("Invalid stdout")?;
+    let stdout_json: serde_json::Value = serde_json::from_str(stdout_str)?;
+
+
+    if let Some(global_env) = stdout_json["env"].as_object() {
+        for (key, value) in process_env(&serde_json::Value::Object(global_env.clone())) {
+            std::env::set_var(key, value.trim_matches('"'));
+        }
+    }
+
+    if let Some(app_env) = stdout_json["apps"][app]["env"].as_object() {
+        for (key, value) in process_env(&serde_json::Value::Object(app_env.clone())) {
+            std::env::set_var(key, value.trim_matches('"'));
+        }
+    }
+
+    Ok(())
+}
+
+fn source_env_file(file_path: &str) -> std::io::Result<HashMap<String, String>> {
+    let path = Path::new(file_path);
+
+    if !path.exists() {
+        eprintln!("File does not exist: {}", file_path);
+        return Err(std::io::Error::new(std::io::ErrorKind::NotFound, "File does not exist"));
+    }
+
+    if !path.is_file() {
+        eprintln!("File is not readable: {}", file_path);
+        return Err(std::io::Error::new(std::io::ErrorKind::PermissionDenied, "File is not readable"));
+    }
+
+    let file = std::fs::File::open(file_path)?;
+    let reader = std::io::BufReader::new(file);
+
+    let mut env_vars = HashMap::new();
+    for line in reader.lines() {
+        let line = line?;
+        if !line.trim().is_empty() && !line.starts_with('#') {
+            if let Some((key, value)) = line.split_once('=') {
+                env_vars.insert(key.to_string(), value.to_string());
+            }
+        }
```

Comment:
This is taking an inline comment as part of the value.
For example:
```bash
X=val # This is the comment
```
exports `X` with value: `val # This is the comment`.

Please refer [here](https://docs.docker.com/compose/environment-variables/variable-interpolation/#env-file-syntax) for the syntax.

Reply:
Good catch! Fixed thanks!